### PR TITLE
Fix puppet 5.x semantic_puppet API issues

### LINKED ISSUE #79: Needs separate gem dependencies based on Puppet Version
Puppet 5.x comes with a bundled version of the `semantic_puppet` gem that contains new APIs not available in the `semantic_puppet 1.0.0` gem used with Puppet 4.x

These new APIs are preventing Forge module installs if the semantic_puppet gem is installed.

We need to have metadata-json-lint only depend on `semantic_puppet` if `puppet` is < 5.

### PR DESCRIPTION
**Pull request by dhollinger**
Fixes #79

### CODE REVIEW

File: metadata-json-lint.gemspec [hunk 1]
```diff
@@ -19,8 +19,16 @@ Gem::Specification.new do |s|
...
   s.add_development_dependency 'rake'
   s.add_development_dependency 'rspec'
   s.add_development_dependency 'rubocop'
+  s.requirements << 'semantic_puppet >= 0.1.2 < 2.0.0'
+  s.post_install_message = <<-EOS
+  -------------------------------------------------------------------
+      If your puppet is <= 4.x, then the semantic_puppet gem MUST
+      be in your Gemfile!!
+
+      On puppet >= 5.x, semantic_puppet will break functionality!
```
Comment (rnelson0) [inline metadata-json-lint.gemspec]: How about a README update showing what should be added to the Gemfile for Puppet 4?

Comment (dhollinger) [inline metadata-json-lint.gemspec]: Will do

Comment (dhollinger) [inline metadata-json-lint.gemspec]: updated

**Comment by dhollinger** [comment]
@rnelson0 Fixups have been squashed

**Pull request merged**